targets/XPlatCppSdk/source/code/source/playfab/PlayFabWinHttpPlugin.cpp

```diff
@@ -0,0 +1,378 @@
+#include <stdafx.h>
+
+#include <playfab/PlayFabWinHttpPlugin.h>
+#include <playfab/PlayFabSettings.h>
+#include <stdexcept>
+#include <vector>
+#include <windows.h>
+
+#pragma warning (disable: 4100) // formal parameters are part of a public interface
+#pragma warning (disable: 4245) // Some DWORD arguments of WinHTTP API can accept -1, according to documentation
+
+namespace PlayFab
+{
+    PlayFabWinHttpPlugin::PlayFabWinHttpPlugin()
+    {
+        activeRequestCount = 0;
+        threadRunning = true;
+        workerThread = std::thread(&PlayFabWinHttpPlugin::WorkerThread, this);
+    };
+
+    PlayFabWinHttpPlugin::~PlayFabWinHttpPlugin()
+    {
+        threadRunning = false;
+        workerThread.join();
```

Comment:
>  workerThread.join(); [](start = 6, length = 22)

this might throw if the thread is not joinable

Reply:
I agree, but it won't be not joinable because we control all 3 conditions of it:
[http://www.cplusplus.com/reference/thread/thread/joinable/](http://www.cplusplus.com/reference/thread/thread/joinable/)

---
In reply to: [244876721](https://github.com/PlayFab/SDKGenerator/pull/368#discussion_r244876721) [](ancestors = 244876721)

Reply:
Yes - now we do -- we're not sure how the code will morph over time - better be safe than sorry :)

---
In reply to: [244881236](https://github.com/PlayFab/SDKGenerator/pull/368#discussion_r244881236) [](ancestors = 244881236,244876721)